pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java

```diff
@@ -454,6 +427,63 @@ private void setAggregationResults(BrokerResponseNative brokerResponseNative,
     }
   }
 
+  private void setDistinctQueryResults(BrokerRequest brokerRequest, BrokerResponseNative brokerResponseNative,
```

Comment:
Will need to adapt to https://github.com/apache/incubator-pinot/pull/4833.

Reply:
That seems like a pretty big PR which is still under review. If people agree, may be we can merge this. What do you think @npawar ?

Reply:
A small change will be needed in setDistinctResults() method in BrokerReduceService